saleor/static/dashboard-next/shipping/views/ShippingZoneDetails/index.tsx

```diff
@@ -0,0 +1,513 @@
+                                path={shippingZoneRatePath(
+                                  ":shippingZoneId",
+                                  ":rateId"
+                                )}
+                                render={({ match }) => {
+                                  const rate = getShippingMethod(
+                                    decodeURIComponent(match.params.rateId)
+                                  );
+                                  return (
+                                    <ShippingZoneRateDialog
+                                      action="edit"
+                                      confirmButtonState={
+                                        updateRateTransitionState
+                                      }
+                                      defaultCurrency={maybe(
+                                        () => shop.defaultCurrency
+                                      )}
+                                      disabled={
+                                        ops.shippingRateUpdate.opts.loading
+                                      }
+                                      onClose={closeModal}
+                                      onSubmit={formData =>
+                                        ops.shippingRateUpdate.mutate({
+                                          id: decodeURIComponent(
+                                            match.params.rateId
+                                          ),
+                                          input: {
+                                            maximumOrderPrice:
+                                              formData.maxValue,
+                                            minimumOrderPrice:
+                                              formData.minValue,
+                                            name: formData.name,
+                                            price: formData.isFree
+                                              ? 0
+                                              : formData.price,
+                                            shippingZone: id,
+                                            type: maybe(() => rate.type)
+                                          }
+                                        })
+                                      }
+                                      open={!!match}
+                                      rate={rate}
+                                      variant={maybe(() => rate.type)}
+                                    />
+                                  );
+                                }}
+                              />
+                            </Switch>
+                          </>
+                        );
+                      }}
+                    </TypedShippingZone>
+                  )}
+                </ShippingZoneOperations>
+              );
+            }}
+          </Shop>
+        )}
+      </Messages>
+    )}
+  </Navigator>
+);
```

Comment:
IMO this component is too big and not realy readable. Consider splitting/cleaning it.